resource-aggregate/service/service_test.go

```diff
@@ -3,93 +3,66 @@ package service_test
+	config.APIs.GRPC.Server.Addr = "localhost:9888"
+	config.APIs.GRPC.Capabilities.SnapshotThreshold = 1
 
 	oauthShutdown := oauthTest.SetUp(t)
 	defer oauthShutdown()
 
-	ctx := kitNetGrpc.CtxWithToken(context.Background(), oauthTest.GetServiceToken(t))
-
 	authShutdown := authService.SetUp(t)
 	defer authShutdown()
 
-	config.Service.AuthServerAddr = testCfg.AUTH_HOST
-	config.Service.JwksURL = testCfg.JWKS_URL
-	config.Service.OAuth.ClientID = testCfg.OAUTH_MANAGER_CLIENT_ID
-	config.Service.OAuth.Endpoint.TokenURL = testCfg.OAUTH_MANAGER_ENDPOINT_TOKENURL
-	config.Service.UserDevicesManagerTickFrequency = time.Millisecond * 500
-	config.Service.UserDevicesManagerExpiration = time.Millisecond * 500
-
-	clientCertManager, err := certManager.NewCertManager(config.Dial)
-	require.NoError(t, err)
-	dialTLSConfig := clientCertManager.GetClientTLSConfig()
+	raShutdown := test.New(t, config)
+	defer raShutdown()
 
-	eventstore, err := mongodb.NewEventStore(ctx, config.MongoDB, nil, mongodb.WithTLS(dialTLSConfig))
-	require.NoError(t, err)
-	defer eventstore.Clear(ctx)
-
-	config.Service.Addr = "localhost:9888"
-	config.Service.SnapshotThreshold = 1
-
-	server, err := refImpl.Init(config)
-	require.NoError(t, err)
-	defer server.Shutdown()
-	go func() {
-		err := server.Serve()
-		require.NoError(t, err)
-	}()
+	ctx := kitNetGrpc.CtxWithToken(context.Background(), oauthTest.GetServiceToken(t))
 
-	authConn, err := grpc.Dial(config.Service.AuthServerAddr, grpc.WithTransportCredentials(credentials.NewTLS(dialTLSConfig)))
+	authConn, err := client.New(testCfg.MakeGrpcClientConfig(config.Clients.AuthServer.Addr), log.Get().Desugar())
 	require.NoError(t, err)
-	authClient := pbAS.NewAuthorizationServiceClient(authConn)
+	defer authConn.Close()
+	authClient := pbAS.NewAuthorizationServiceClient(authConn.GRPC())
 
-	raConn, err := grpc.Dial(config.Service.Addr, grpc.WithTransportCredentials(credentials.NewTLS(dialTLSConfig)))
+	raConn, err := client.New(testCfg.MakeGrpcClientConfig(config.APIs.GRPC.Server.Addr), log.Get().Desugar())
 	require.NoError(t, err)
-	raClient := service.NewResourceAggregateClient(raConn)
+	defer raConn.Close()
+	raClient := service.NewResourceAggregateClient(raConn.GRPC())
 
 	deviceId := "dev0"
 	href := "/oic/p"
 	code := oauthTest.GetDeviceAuthorizationCode(t)
-	resp, err := authClient.SignUp(ctx, &pbAS.SignUpRequest{
+	_, err = authClient.SignUp(ctx, &pbAS.SignUpRequest{
 		DeviceId:              deviceId,
 		AuthorizationCode:     code,
 		AuthorizationProvider: "plgd",
 	})
 	require.NoError(t, err)
 
-	pubReq := testMakePublishResourceRequest(deviceId, href, resp.UserId, resp.AccessToken)
+	pubReq := testMakePublishResourceRequest(deviceId, []string{href})
 	_, err = raClient.PublishResourceLinks(ctx, pubReq)
 	require.NoError(t, err)
 
-	unpubReq := testMakeUnpublishResourceRequest(deviceId, href, resp.UserId, resp.AccessToken)
+	unpubReq := testMakeUnpublishResourceRequest(deviceId, []string{href})
```

Comment:
Why did you change in config refactoring args of test func ?

Reply:
I merged test functions.